File: penglings.m
file = readtable('penglings.csv');
Adelie = table2array(file(1:151,4:7));
Gentoo = table2array(file(152:274,4:7));
Chinstrap = table2array(file(275:342,4:7));

b_Adelie = Adelie(:,1);
f_Adelie = Adelie(:,3);
m_Adelie = Adelie(:,4);
b_Gentoo = Gentoo(:,1);
f_Gentoo = Gentoo(:,3);
m_Gentoo = Gentoo(:,4);
b_Chinstrap = Chinstrap(:,1);
f_Chinstrap = Chinstrap(:,3);
m_Chinstrap = Chinstrap(:,4);

clf;
figure = bubblechart(f_Adelie,m_Adelie,b_Adelie-30,'#FF8C00','MarkerFaceAlpha',0.8);
bubblesize([1 30]);
hold on;
figure = bubblechart(f_Gentoo,m_Gentoo,b_Gentoo-30,'#008B8B','MarkerFaceAlpha',0.8);
bubblesize([1 30]);
hold on;
figure = bubblechart(f_Chinstrap,m_Chinstrap,b_Chinstrap-30,'#9932CC','MarkerFaceAlpha',0.8);
bubblesize([1 30]);
xlabel("Flipper Length (mm)");
ylabel("Body Mass (g)");
hold off;

File: penglings.py
import matplotlib.pyplot as plt
import csv

pengdict = {
    'Adelie':{
        'bill_length_mm':[],
        'flipper_length_mm':[],
        'body_mass_g':[]
    },
    'Gentoo':{
        'bill_length_mm':[],
        'flipper_length_mm':[],
        'body_mass_g':[]
    },
    'Chinstrap':{
        'bill_length_mm':[],
        'flipper_length_mm':[],
        'body_mass_g':[]
    }
}
with open('penglings.csv', newline='') as file:
    reader = csv.DictReader(file)
    for entry in reader:
        if (entry['bill_length_mm'] != 'NA'):
            pengdict[entry['species']]['bill_length_mm'].append(((float(entry['bill_length_mm'])-20))**2)
            pengdict[entry['species']]['flipper_length_mm'].append(float(entry['flipper_length_mm']))
            pengdict[entry['species']]['body_mass_g'].append(float(entry['body_mass_g']))

fig = plt.figure(figsize=(14,8))
plt.scatter(pengdict['Adelie']['flipper_length_mm'], pengdict['Adelie']['body_mass_g'], s=pengdict['Adelie']['bill_length_mm'], c='#FF8C00', alpha=0.8)
plt.scatter(pengdict['Gentoo']['flipper_length_mm'], pengdict['Gentoo']['body_mass_g'], s=pengdict['Gentoo']['bill_length_mm'], c='#008B8B', alpha=0.8)
plt.scatter(pengdict['Chinstrap']['flipper_length_mm'], pengdict['Chinstrap']['body_mass_g'], s=pengdict['Chinstrap']['bill_length_mm'], c='#9932CC', alpha=0.8)
plt.xlabel("Flipper Length (mm)")
plt.ylabel("Body Mass (g)")

plt.show()
plt.savefig('pythonpenglings.png')